### PR TITLE
Add -hidden_activation argument for configurable hidden-layer activation

### PR DESCRIPTION
**Pull request by StoryofLegend**
## Summary
Adds an optional `-hidden_activation` argument to `GenNet.py train` that lets the user override the activation function used in the hidden layers. When ommitted, original behaviour is preserved: tanh for classification, relu for regression

## Motivation
Making it configurable is useful for methodological comparisons, e.g. studying how the activation choice affects feature importance / interpretability and the stability of feature rankings.

## Behaviour
- When `-hidden_activation` is omitted, behaviour is unchanged: `tanh` for classification, `relu` for regression.
- When provided, the value is passed directly to `keras.layers.Activation`, so any valid Keras activation string works (e.g. `relu`, `tanh`, `sigmoid`, etc.).

### Example
python GenNet.py train -path ./data -out ./results -ID 1 -problem_type classification -hidden_activation relu

Add -hidden_activation argument for configurable hidden layer activation. Allow user to override the default activation function for hidden layers via -hidden_activation <keras activation string>. When ommitted, original behaviour is preserved: tanh for classification, relu for regression

### CODE REVIEW

**Comment by ArnovanHilten** [comment]
Looks good, thank you for your contribution!
